src/TypeSpec.Extension/Emitter.Csharp/src/lib/clientModelBuilder.ts

```diff
@@ -86,59 +86,51 @@ export function createModelForService(
         context,
         "@azure-tools/typespec-csharp"
     );
-    const title = service.title;
     const serviceNamespaceType = service.type;
-    const apiVersions: Set<string> = new Set<string>();
-    let version = service.version;
-    if (version && version !== mockApiVersion) {
-        apiVersions.add(version);
-    }
+
+    const apiVersions: Set<string> | undefined = new Set<string>();
+    let defaultApiVersion: string | undefined = undefined;
     const versions = getVersions(program, service.type)[1]?.getVersions();
     if (versions) {
         for (const ver of versions) {
             apiVersions.add(ver.value);
         }
-        version = versions[versions.length - 1].value; //default version
+        defaultApiVersion = versions[versions.length - 1].value;
```

Comment:
Better to check the count of versions. in case, service defined @versioned, but the enum is empty. Add the check to avoid crash.

Reply:
If the enum is empty, compiler will throw error.

Reply:
I don't think typespec compiler will throw error when the enum is empty.